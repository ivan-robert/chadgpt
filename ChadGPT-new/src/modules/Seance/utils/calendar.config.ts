import { LocaleConfig } from "react-native-calendars";

export const configCalendar = () => {
  LocaleConfig.locales["fr"] = {
    monthNames: [
      "Janvier",
      "Février",
      "Mars",
      "Avril",
      "Mai",
      "Juin",
      "Juillet",
      "Août",
      "Septembre",
      "Octobre",
      "Novembre",
      "Décembre",
    ],
    monthNamesShort: [
      "Janv.",
      "Févr.",
      "Mars",
      "Avril",
      "Mai",
      "Juin",
      "Juil.",
      "Août",
      "Sept.",
      "Oct.",
      "Nov.",
      "Déc.",
    ],
    dayNames: [
      "Dimanche",
      "Lundi",
      "Mardi",
      "Mercredi",
      "Jeudi",
      "Vendredi",
      "Samedi",
    ],
    dayNamesShort: ["Dim.", "Lun.", "Mar.", "Mer.", "Jeu.", "Ven.", "Sam."],
    today: "Aujourd'hui",
  };

  LocaleConfig.locales["en"] = {
    monthNames: [
      "January",
      "February",
      "March",
      "April",
      "May",
      "June",
      "July",
      "August",
      "September",
      "October",
      "November",
      "December",
    ],
    monthNamesShort: [
      "Jan.",
      "Feb.",
      "Mar.",
      "Apr.",
      "May",
      "Jun.",
      "Jul.",
      "Aug.",
      "Sep.",
      "Oct.",
      "Nov.",
      "Dec.",
    ],
    dayNames: [
      "Sunday",
      "Monday",
      "Tuesday",
      "Wednesday",
      "Thursday",
      "Friday",
      "Saturday",
    ],
    dayNamesShort: ["Sun.", "Mon.", "Tue.", "Wed.", "Thu.", "Fri.", "Sat."],
    today: "Today",
  };
};
