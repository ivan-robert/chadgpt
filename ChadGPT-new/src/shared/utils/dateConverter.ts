import { DateTime } from "luxon";

export function timestampzToDate(timestampz: string): Date {
  return new Date(timestampz);
}

export function dateToTimestampz(date: Date): string {
  return date.toISOString();
}

export function dateToDateString(date: Date): string {
  return DateTime.fromJSDate(date).toFormat("dd-MM-yyyy");
}

export function dateStringToDate(dateString: string): Date {
  return new Date(dateString);
}
