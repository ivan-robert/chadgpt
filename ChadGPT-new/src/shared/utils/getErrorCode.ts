export const getErrorCode = (error: unknown): string | null => {
  if (error === null) {
    return null;
  }
  if (typeof error !== "object") {
    return null;
  }

  if ("code" in error) {
    return JSON.stringify(error["code"]);
  }
  return null;
};
