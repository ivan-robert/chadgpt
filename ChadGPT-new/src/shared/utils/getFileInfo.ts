import { SupabaseClient } from "@supabase/supabase-js";

export const getFileLinkFromBucket = (
  supabaseClient: SupabaseClient,
  bucket: string,
  folder: string,
  name: string
) => {
  return supabaseClient.storage
    .from(bucket)
    .getPublicUrl(`${folder}${folder.length ? "/" : ""}${name}`).data.publicUrl;
};
