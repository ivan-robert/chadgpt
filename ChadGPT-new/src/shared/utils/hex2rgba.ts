export function hex2rgba(hex: string, alpha = 1) {
  hex = hex.replace(/^#/, "");

  if (hex.length === 3) {
    hex = hex
      .split("")
      .map((char) => char + char)
      .join("");
  }

  const r = parseInt(hex.slice(0, 2), 16);
  const g = parseInt(hex.slice(2, 4), 16);
  const b = parseInt(hex.slice(4, 6), 16);

  let a = alpha;
  if (hex.length === 8) {
    a = parseInt(hex.slice(6, 8), 16) / 255;
  }

  return `rgba(${r}, ${g}, ${b}, ${a})`;
}
