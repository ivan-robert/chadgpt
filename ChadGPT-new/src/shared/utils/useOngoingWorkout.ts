import { GenericPerf, perfSchema } from "#modules/Seance/domain/perf.types";
import {
  PreparedWorkout,
  preparedWorkoutSchema,
  Workout,
  workoutSchema,
} from "#modules/Seance/domain/seance.types";
import { LocalStorage } from "#shared/service/storage/Storage.service";
import { ToastService } from "#shared/service/Toast.service";
import { workoutAtom } from "#shared/store/ongoingWorkout";
import { useSetAtom } from "jotai";

export const retrieveOngoingWorkout = (): Omit<
  Workout,
  "endDate" | "salle"
> | null => {
  const workout = LocalStorage.getStringItem("workout");
  if (!workout) return null;
  const parsedWorkout = JSON.parse(workout);

  const isWorkoutValid = workoutSchema
    .omit({ endDate: true, salle: true })
    .safeParse(parsedWorkout).success;

  if (!isWorkoutValid) {
    ToastService.show({
      type: "error",
      message:
        "We encountered an error while retrieving your workout. Please contact support if this happens again.",
      title: "Error",
    });
    return null;
  }

  if (!parsedWorkout.perfs?.length) {
    return null;
  }

  return parsedWorkout;
};

const saveOngoingWorkout = (workout: Workout) => {
  LocalStorage.setStringItem("workout", JSON.stringify(workout));
};

const retrieveCurrentTemplate = (): PreparedWorkout | null => {
  const workout = LocalStorage.getStringItem("current_workout_template");
  if (!workout) return null;
  const parsedWorkout = JSON.parse(workout);

  if (
    !parsedWorkout ||
    !preparedWorkoutSchema.safeParse(JSON.parse(workout)).success
  ) {
    return null;
  }

  return parsedWorkout;
};

const saveCurrentTemplate = (template: PreparedWorkout) => {
  LocalStorage.setStringItem(
    "current_workout_template",
    JSON.stringify(template)
  );
};

const retrieveOngoingPerf = (): GenericPerf | null => {
  const perf = LocalStorage.getStringItem("ongoing_perf");
  if (!perf) return null;
  const rawPerf = JSON.parse(perf);
  const { success } = perfSchema.safeParse(rawPerf);
  if (!success) return null;
  return rawPerf;
};

const saveOngoingPerf = (perf: GenericPerf) => {
  LocalStorage.setStringItem("ongoing_perf", JSON.stringify(perf));
};

const removeOngoingPerf = () => {
  LocalStorage.removeItem("ongoing_perf");
};

export const useOngoingWorkout = () => {
  const setOngoingWorkout = useSetAtom(workoutAtom);
  const removeOngoingWorkout = () => {
    LocalStorage.removeItem("workout");
    LocalStorage.removeItem("ongoing_perf");
    LocalStorage.removeItem("current_workout_template");

    setOngoingWorkout({ isOngoing: false });
  };
  const renderOngoingWorkout = () => {
    setOngoingWorkout((prev) => ({ ...prev }));
  };
  return {
    removeOngoingWorkout,
    retrieveOngoingWorkout,
    retrieveOngoingPerf,
    saveOngoingPerf,
    removeOngoingPerf,
    renderOngoingWorkout,
    retrieveCurrentTemplate,
    saveOngoingWorkout,
    saveCurrentTemplate,
  };
};
