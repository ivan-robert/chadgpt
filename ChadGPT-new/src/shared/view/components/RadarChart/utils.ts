const degToRadians = (degree: number): number => (degree * Math.PI) / 180;

export type Point = [number, number];

export const edgePointCalculatorFor =
  (center: Point, radius: number) =>
  (degree: number, radiusPercentage = 100): Point => {
    const degreeInRadians = degToRadians(degree);
    return [
      center[0] + (Math.cos(degreeInRadians) * radius * radiusPercentage) / 100,
      center[1] + (Math.sin(degreeInRadians) * radius * radiusPercentage) / 100,
    ];
  };
