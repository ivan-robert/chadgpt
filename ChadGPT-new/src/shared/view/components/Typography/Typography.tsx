import React, { ReactNode, forwardRef } from "react";
import { Text as RNText, TextProps } from "react-native";
import { Color } from "../../../../theme/colors.types";
import { FontStyle } from "../../../../theme/fonts.types";
import { useTheme } from "@emotion/react";

type TypographyProps = TextProps & {
  children?: ReactNode;
  color?: Color;
  isUnderlined?: boolean;
  maxLinesBeforeEllipsis?: number;
};

type TextSize = "l" | "m" | "s";
type HeadlineSize = "l" | "xl";
type TitleSize = "l" | "m";

const BaseText = forwardRef<RNText, TypographyProps>(
  (
    {
      children,
      color,
      isUnderlined,
      maxLinesBeforeEllipsis,
      style,
      ...textProps
    },
    ref
  ) => {
    const theme = useTheme();
    return (
      <RNText
        ref={ref}
        style={[
          { color: color ?? theme.colors.text200 },
          {
            textDecorationLine: isUnderlined ? "underline" : "none",
          },
          style,
        ]}
        numberOfLines={maxLinesBeforeEllipsis}
        {...textProps}
      >
        {children}
      </RNText>
    );
  }
);

const Title = (size: TitleSize, fontStyle: FontStyle) =>
  React.forwardRef<RNText, TypographyProps>(
    (
      {
        accessibilityRole = "header",
        color = "black",
        style: textStyle = {},
        ...props
      },
      ref
    ) => {
      const theme = useTheme();
      return (
        <BaseText
          ref={ref}
          color={color}
          {...props}
          //eslint-disable-next-line
          //@ts-ignore
          style={{
            //@ts-expect-error Somehow now the style can be falsy
            ...textStyle,
            ...theme.typography.title[size][fontStyle],
          }}
          accessibilityRole={accessibilityRole}
        />
      );
    }
  );

const Headline = (size: HeadlineSize, fontStyle: FontStyle) =>
  React.forwardRef<RNText, TypographyProps>(
    (
      {
        accessibilityRole = "header",
        color = "black",
        style: textStyle = {},
        ...props
      },
      ref
    ) => {
      const theme = useTheme();

      return (
        <BaseText
          ref={ref}
          color={color}
          style={{
            ...theme.typography.headline[size][fontStyle],
            //eslint-disable-next-line
            //@ts-ignore
            ...textStyle,
          }}
          {...props}
          accessibilityRole={accessibilityRole}
        />
      );
    }
  );

const Text = (size: TextSize, fontStyle: FontStyle) =>
  React.forwardRef<RNText, TypographyProps>(
    (
      {
        accessibilityRole = "header",
        color = "black",
        style: textStyle = {},
        ...props
      },
      ref
    ) => {
      const theme = useTheme();
      return (
        <BaseText
          ref={ref}
          color={color}
          {...props}
          //eslint-disable-next-line
          //@ts-ignore
          style={{
            ...theme.typography.text[size][fontStyle],
            flexShrink: 1,
            //@ts-expect-error Somehow now the style can be falsy
            ...textStyle,
          }}
          accessibilityRole={accessibilityRole}
        />
      );
    }
  );

//eslint-disable-next-line
//@ts-ignore

export const Typography = {
  TitleL: {
    regular: Title("l", "regular"),
    bold: Title("l", "bold"),
    italic: Title("l", "italic"),
  },
  TitleM: {
    regular: Title("m", "regular"),
    bold: Title("m", "bold"),
    italic: Title("m", "italic"),
  },
  HeadlineL: {
    regular: Headline("l", "regular"),
    bold: Headline("l", "bold"),
    italic: Headline("l", "italic"),
  },
  HeadlineXL: {
    regular: Headline("xl", "regular"),
    bold: Headline("xl", "bold"),
    italic: Headline("xl", "italic"),
  },
  TextL: {
    regular: Text("l", "regular"),
    bold: Text("l", "bold"),
    italic: Text("l", "italic"),
  },
  TextM: {
    regular: Text("m", "regular"),
    bold: Text("m", "bold"),
    italic: Text("m", "italic"),
  },
  TextS: {
    regular: Text("s", "regular"),
    bold: Text("s", "bold"),
    italic: Text("s", "italic"),
  },
};
