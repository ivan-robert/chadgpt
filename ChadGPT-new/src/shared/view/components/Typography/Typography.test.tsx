import { screen } from "@testing-library/react-native";

import { Typography } from "./Typography";
import { renderWithProviders } from "../../../../../jest/utils/UtilFunctions";

describe("Typography", () => {
  test("Title xlarge matches snapshot", () => {
    renderWithProviders(
      <Typography.HeadlineL.bold>hello</Typography.HeadlineL.bold>
    );

    expect(screen).toMatchSnapshot();
  });
});
