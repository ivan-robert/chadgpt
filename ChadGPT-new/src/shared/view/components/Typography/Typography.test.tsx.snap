// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Title xlarge matches snapshot 1`] = `
<Text
  accessibilityRole="header"
  style={
    [
      {
        "color": "black",
      },
      {
        "textDecorationLine": "none",
      },
      {
        "fontFamily": "Montserrat-Bold",
        "fontSize": 40,
        "lineHeight": 48,
      },
    ]
  }
>
  hello
</Text>
`;
